app/vmui/packages/vmui/src/components/Table/TableSettings/TableSettings.tsx

```diff
@@ -100,42 +128,64 @@ const TableSettings: FC<TableSettingsProps> = ({
             />
           </div>
           <div className="vm-table-settings-popper-list">
-            <div className="vm-table-settings-popper-list-header">
-              <h3 className="vm-table-settings-popper-list-header__title">Display columns</h3>
-              <Tooltip title="Reset to default">
-                <Button
-                  color="primary"
-                  variant="text"
-                  size="small"
-                  onClick={handleResetColumns}
-                  startIcon={<RestartIcon/>}
-                  ariaLabel="reset columns"
-                />
-              </Tooltip>
-            </div>
-            <div
-              className="vm-table-settings-popper-list__item vm-table-settings-popper-list__check_all"
-            >
-              <Checkbox
-                checked={defaultColumns.length === columns.length}
-                onChange={toggleAllColumns}
-                label="Check all"
-                disabled={tableCompact}
-              />
-            </div>
-            {columns.map(col => (
-              <div
-                className="vm-table-settings-popper-list__item"
-                key={col}
-              >
-                <Checkbox
-                  checked={defaultColumns.includes(col)}
-                  onChange={createHandlerChange(col)}
-                  label={col}
-                  disabled={tableCompact}
+            <div>
+              <div className="vm-table-settings-popper-list-header">
+                <h3 className="vm-table-settings-popper-list-header__title">Display columns</h3>
+                <Tooltip title="search column">
+                  <Button
+                    color="primary"
+                    variant="text"
+                    onClick={toggleShowSearch}
+                    startIcon={<SearchIcon/>}
+                    ariaLabel="reset columns"
+                  />
+                </Tooltip>
+              </div>
+              {showSearch && (
+                <TextField
+                  placeholder={"search column"}
+                  startIcon={<SearchIcon/>}
+                  value={searchColumn}
+                  onChange={setSearchColumn}
+                  onBlur={handleBlurSearch}
+                  onKeyDown={handleKeyDown}
+                  type="search"
                 />
+              )}
+              {!filteredColumns.length && (
+                <p className="vm-table-settings-popper-list__no-found">No columns found</p>
+              )}
+              <div className="vm-table-settings-popper-list-header">
+                {!!filteredColumns.length && (
+                  <div className="vm-table-settings-popper-list__item vm-table-settings-popper-list__item_check_all">
+                    <Checkbox
+                      checked={defaultColumns.length === columns.length}
```

Comment:
Looks like this button should check or uncheck only filtered fields.
For example, if I filtered fields by search keyword and pressed check-all, I'd expect only the fields I see in the options to be checked. In fact, it checks all the fields.